compiler/rustc_codegen_ssa/src/mir/mod.rs

```diff
@@ -293,21 +303,75 @@ pub fn codegen_mir<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
     // So drop the builder of `start_llbb` to avoid having two at the same time.
     drop(start_bx);
 
-    let mut unreached_blocks = DenseBitSet::new_filled(mir.basic_blocks.len());
     // Codegen the body of each reachable block using our reverse postorder list.
     for bb in traversal_order {
         fx.codegen_block(bb);
-        unreached_blocks.remove(bb);
     }
+}
+
+// FIXME: Move this function to mir::transform when post-mono MIR passes land.
+fn optimize_use_clone<'tcx>(tcx: TyCtxt<'tcx>, mut mir: Body<'tcx>) -> Body<'tcx> {
+    if tcx.features().ergonomic_clones() {
+        for bb in mir.basic_blocks.as_mut() {
+            let mir::TerminatorKind::Call {
+                func,
+                args,
+                destination,
+                target,
+                call_source: mir::CallSource::Use,
+                ..
+            } = &bb.terminator().kind
+            else {
+                continue;
+            };
+
+            // It's definitely not a clone if there are multiple arguments
+            let [arg] = &args[..] else { continue };
+
+            // These types are easily available from locals, so check that before
+            // doing DefId lookups to figure out what we're actually calling.
+            let arg_ty = arg.node.ty(&mir.local_decls, tcx);
+
+            // monomorphize
+            // is already monomorphized
+            // let arg_ty = instance.instantiate_mir_and_normalize_erasing_regions(
+            //     cx.tcx(),
+            //     cx.typing_env(),
+            //     ty::EarlyBinder::bind(arg_ty),
+            // );
```

Comment:
This needs to be removed, the whole mir is monomorphized as soon as we start codegen.